substrate/primitives/staking/src/lib.rs

```diff
@@ -254,6 +254,9 @@ pub trait StakingInterface {
 	/// schedules have reached their unlocking era should allow more calls to this function.
 	fn unbond(stash: &Self::AccountId, value: Self::Balance) -> DispatchResult;
 
+	/// Update the reward destination for the ledger associated with the stash.
+	fn update_payee(stash: &Self::AccountId, reward_acc: &Self::AccountId) -> DispatchResult;
```

Comment:
We want to keep this trait as general as possible, but at the same time it seems that the `update_payee` is too simple in the context of the staking pallet. An accountId is not powerful enough to express the `RewardDestination` type used in our staking system. Maybe we could parameterise this method with a generic that can be used to express the reward destination in the staking pallet?

Reply:
Reiterating, `RewardDestination` is an implementation detail and isn't really necessary in this context of a staking interface.

Let's run a thought experiment. Say we bring `RewardDestination` to the staking primitives to allow for more general uses of functions like this. It is an interface anyone can implement, it's not constrained to `pallet-staking`. What if users implementing this interface have additional rewards destinations other than what we currently use in `pallet-staking`? What if they use just one destination out of the several we provide? We just constrained them in this interface to what we are using in `pallet-staking`.

In this example, let's switch the approach and make this an associated type. That doesn't help right now because we don't use other types. If there are users out there that would want other reward destinations, they can just extend `StakingInterface` themselves with that additional type or an additional function and bake in any extra logic. It's what we are doing with `StakingUnsafe` (pending rename) right now :grin:.

We should aim at keeping this interface as simple as possible while enabling anyone who implements it to interact with other substrate-based staking consumers. My personal feeling is that the function right now is a good balance of functionality and ergonomics.